plugins/uefi/efi/fwupdate.c

```diff
@@ -408,7 +408,7 @@ fwup_delete_boot_entry(VOID)
 		/* check if the variable name is Boot#### */
 		if (CompareGuid(&vendor_guid, &global_variable_guid) != 0)
 			continue;
-		if (StrCmp(variable_name, L"Boot") != 0)
+		if ((StrLen(variable_name) != 8) || (StrnCmp(variable_name, L"Boot", 4) != 0))
 			continue;
```

Comment:
I think I'd rather have this as:

    if (StrLen(variable_name) != 8)
      continue;
    if (StrnCmp(variable_name, L"Boot", 4) != 0)
      continue

Reply:
Looks good to me.